modules/renter/hostdb/online.go

```diff
@@ -0,0 +1,39 @@
+package hostdb
+
+import (
+	"time"
+)
+
+// online.go performs regular checks against the gateway to see if there is
+// internet connectivity. The check is performed by looking for at least one
+// non-local peer in the peer list. If there is no internet connectivity, scans
+// must be stopped lest we penalize otherwise online hosts.
+
+func (hdb *HostDB) threadedOnlineCheck() {
+	err := hdb.tg.Add()
+	if err != nil {
+		return
+	}
+	defer hdb.tg.Done()
+
+	for {
+		// Every 30 seconds, check the online status and update the online
+		// field.
+		peers := hdb.gateway.Peers()
```

Comment:
I think we are going to want to add a `gateway.HasInternetConnection` method. The gateway itself needs such a method because otherwise it will prune nodes while offline (I think -- there may already be protections against this). Can be done in a separate PR though